sirepo/raydata/analysis_driver/chx.py

```diff
@@ -0,0 +1,54 @@
+"""Analysis specific to CHX
+
+:copyright: Copyright (c) 2023 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern import pkconfig
+from pykern import pkio
+from pykern.pkcollections import PKDict
+from pykern.pkdebug import pkdp
+import sirepo.raydata.analysis_driver
+
+_cfg = None
+
+
+class CHX(sirepo.raydata.analysis_driver.AnalysisDriverBase):
+    def get_notebooks(self):
+        return [
+            PKDict(
+                input_f=_cfg.base_dir.join(
+                    self._scan_metadata.get_start_field("cycle"),
+                    "AutoRuns",
+                    self._scan_metadata.get_start_field("user"),
+                    f"{self._scan_metadata.get_start_field('auto_pipeline')}.ipynb",
+                ),
+                output_f=_cfg.base_dir.join(
+                    self._scan_metadata.get_start_field("cycle"),
+                    self._scan_metadata.get_start_field("user"),
+                    "ResPipelines",
+                    f"{self._scan_metadata.get_start_field('auto_pipeline')}_{self.uid}.ipynb",
+                ),
+            )
+        ]
+
+    def _get_papermill_args(self, *args, **kwargs):
+        return [
+            ["run_two_time", True],
+            ["run_dose", False],
+            ["username", self._scan_metadata.get_start_field("user")],
+        ]
+
+    def get_output_dir(self):
+        return _cfg.base_dir.join(
+            self._scan_metadata.get_start_field("cycle"),
+            self._scan_metadata.get_start_field("user"),
+            "Results",
+            self.uid,
+        )
```

Comment:
I saw this error in the analysis log when trying to Run Analysis on a scan. Do we need to create this directory? Or is this something the notebooks should handle?

```python
Executing:   0%|          | 0/371 [00:00<?, ?cell/s]Traceback (most recent call last):
  File "/home/vagrant/.pyenv/versions/py3/bin/papermill", line 8, in <module>
    sys.exit(papermill())
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/click/decorators.py", line 33, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/cli.py", line 250, in papermill
    execute_notebook(
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/execute.py", line 113, in execute_notebook
    nb = papermill_engines.execute_notebook_with_engine(
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/engines.py", line 49, in execute_notebook_with_engine
    return self.get_engine(engine_name).execute_notebook(nb, kernel_name, **kwargs)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/engines.py", line 365, in execute_notebook
    nb_man.notebook_start()
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/engines.py", line 77, in wrapper
    return func(self, *args, **kwargs)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/engines.py", line 206, in notebook_start
    self.save()
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/engines.py", line 77, in wrapper
    return func(self, *args, **kwargs)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/engines.py", line 147, in save
    write_ipynb(self.nb, self.output_path)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/iorw.py", line 499, in write_ipynb
    papermill_io.write(nbformat.writes(nb), path)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/iorw.py", line 106, in write
    return self.get_handler(path, extensions).write(buf, path)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/papermill/iorw.py", line 226, in write
    raise FileNotFoundError("output folder {} doesn't exist.".format(dirname))
FileNotFoundError: output folder /home/vagrant/src/radiasoft/raydata/dev_notebooks/chx/2023_2/commissioning/ResPipelines doesn't exist.

Executing:   0%|          | 0/371 [00:00<?, ?cell/s]
```

Reply:
I think we can expect that directory already exists. I updated the raydata PR to include it. Pull it and see if that works.

Reply:
That took care of it